From 7cbe536cfe2ec60120d0bc61be4def4d7105f945 Mon Sep 17 00:00:00 2001
From: Ilya Ledvich <ilya@compulab.co.il>
Date: Fri, 8 Apr 2016 09:16:19 +0300
Subject: [PATCH 20/46] ARM: dts: imx7d: cl-som-imx7: add WiLink8 WLAN support

Add support for TI's WiLink8 WLAN module on SD2 bus.

Signed-off-by: Ilya Ledvich <ilya@compulab.co.il>
---
 arch/arm/boot/dts/imx7d-cl-som-imx7.dts | 57 +++++++++++++++++++++++++++++++++
 1 file changed, 57 insertions(+)

diff --git a/arch/arm/boot/dts/imx7d-cl-som-imx7.dts b/arch/arm/boot/dts/imx7d-cl-som-imx7.dts
index 4495fcc..202d52d 100644
--- a/arch/arm/boot/dts/imx7d-cl-som-imx7.dts
+++ b/arch/arm/boot/dts/imx7d-cl-som-imx7.dts
@@ -48,6 +48,28 @@
 			regulator-max-microvolt = <3300000>;
 			regulator-always-on;
 		};
+
+		rf_pwr_en_reg: regulator@2 {
+			compatible = "regulator-fixed";
+			regulator-name = "rf-pwr-en-regulator";
+			regulator-min-microvolt = <3300000>;
+			regulator-max-microvolt = <3300000>;
+			gpio = <&pca9555 9 GPIO_ACTIVE_HIGH>;
+			startup-delay-us = <70000>;
+			enable-active-high;
+			regulator-always-on;
+		};
+
+		wlan_en_reg: regulator@3 {
+			compatible = "regulator-fixed";
+			regulator-name = "wlan-en-regulator";
+			regulator-min-microvolt = <3300000>;
+			regulator-max-microvolt = <3300000>;
+			gpio = <&pca9555 0 GPIO_ACTIVE_HIGH>;
+			/* WLAN card specific delay */
+			startup-delay-us = <70000>;
+			enable-active-high;
+		};
 	};
 
 	leds {
@@ -198,6 +220,7 @@
 	cl-som-imx7 {
 		pinctrl_hog: hoggrp {
 			fsl,pins = <
+				MX7D_PAD_GPIO1_IO15__GPIO1_IO15			0x34 /* WLAN IRQ */
 				MX7D_PAD_SAI1_TX_SYNC__GPIO6_IO14		0x34 /* Onboard LED */
 			>;
 		};
@@ -288,6 +311,17 @@
 			>;
 		};
 
+		pinctrl_usdhc2: usdhc2grp {
+			fsl,pins = <
+				MX7D_PAD_SD2_CMD__SD2_CMD			0x59
+				MX7D_PAD_SD2_CLK__SD2_CLK			0x19
+				MX7D_PAD_SD2_DATA0__SD2_DATA0			0x59
+				MX7D_PAD_SD2_DATA1__SD2_DATA1			0x59
+				MX7D_PAD_SD2_DATA2__SD2_DATA2			0x59
+				MX7D_PAD_SD2_DATA3__SD2_DATA3			0x59
+			>;
+		};
+
 		pinctrl_usdhc3: usdhc3grp {
 			fsl,pins = <
 				MX7D_PAD_SD3_CMD__SD3_CMD			0x59
@@ -473,6 +507,29 @@
 	status = "okay";
 };
 
+&usdhc2 {
+	pinctrl-names = "default";
+	pinctrl-0 = <&pinctrl_usdhc2>;
+	cd-gpios = <>;
+	wp-gpios = <>;
+	no-1-8-v;
+	keep-power-in-suspend;
+	enable-sdio-wakeup;
+	vmmc-supply = <&wlan_en_reg>;
+	depends = "/regulators/regulator@3";
+	non-removable;
+	cap-power-off-card;
+	status = "okay";
+	#address-cells = <1>;
+	#size-cells = <0>;
+	wlcore: wlcore@0 {
+		compatible = "ti,wl1835";
+		reg = <2>;
+		interrupt-parent = <&gpio1>;
+		interrupts = <15 IRQ_TYPE_LEVEL_HIGH>;
+	};
+};
+
 &usdhc3 {
 	pinctrl-names = "default";
 	pinctrl-0 = <&pinctrl_usdhc3>;
-- 
1.9.1

